### PR TITLE
Fixes for gradient accumulation test

### PR DESCRIPTION
**Pull request by achalddave**
Earlier, we used deepcopy, which seems to copy the tensor python objects, but keeps a pointer to the same tensor, meaning we were not properly testing gradient accumulation. This commit mainly fixes that, using load_state_dict() instead to make a model copy.

Unfortunately, the test fails once we do this, and a few changes are necessary to make it pass:

1. Switch AdamW -> SGD
2. Switch to float32

It's unclear why this was necessary, and solving this is left for a future commit.

### CODE REVIEW

**Comment by achalddave** [comment]
Created https://github.com/mlfoundations/open_lm/issues/126 to keep track of the AdamW + grad accum issue.

**Review by GeorgiosSmyrnis — APPROVED** [review]
LGTM - let's keep track of the issue #126 and see what's going on.
